### PR TITLE
[MDB IGNORE] [IDB IGNORE] Overmap Expansion #4

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

hiding trees component taken from https://github.com/tgstation/TerraGov-Marine-Corps/pull/3528

ports https://github.com/tgstation/tgstation/pull/64592

weather effects from https://github.com/tgstation/TerraGov-Marine-Corps/pull/4645

the gruboid sprites are from https://github.com/Whitesands13/Whitesands/pull/804. Was meant to be a placeholder, but I think I became too attached to the placeholders.

rockplanet storms stolen from https://github.com/tgstation/TerraGov-Marine-Corps/pull/4645

Planetgen by Jackrip in https://github.com/voidcrew/Voidcrew-LRP/pull/325
Many sprites that were stolen are in this pr, and are from but:

[Europa](https://github.com/Yonaguni/EuropaStation), [Virgo](https://github.com/PolarisSS13/Polaris), [Civ13](https://github.com/Civ13/Civ13), [Fallout13](https://github.com/BadDeathclaw/Drymouth-Gulch), [Colonial Marines](https://github.com/cmss13-devs/cmss13)

TODO: specify what exactly was whos. If something you made is in here, please say so. I want to give everyone who made sprites for this proper credit

### @Ketrai @drawsstuff  Randal the Vandal, @triplezeta, @SomeAngryMiner, @zestybastard AND @IndusRobot 
Ketrai made a good half of the icemoon turfs, and the rockplanet turfs, thank you so fucking much for letting me use them
Drawsstuff made the  Lavaland moss sprite, and compiled a list of places with turfs  i could steal from
Randal the Vandal and @IndusRobot made the new ruins, massive thanks to them
Tetra provided emotional support
Any (zesty) made the amazing rock walls
Azlan helped with sprites

Todo: Cave areas are a bit fucked. Unfuck them slightly

## Why It's Good For The Game

Outposts are on the horizion, meaning space trucking is soon. But for now, we really, *really*, need content for players. I think biomegen and new planets fit the bill perfectly. 

lavaland
icemoon
jungle
whitesands
wasteplanet
rockplanet
beach

## Changelog
:cl:
add: Three new planets
add: New planet gen. Most planets now have a biome gen, meaning planets will now have more variance
imageadd: New rock walls by Any%
tweak: Trees now turn transparent if something is behind them.
fix: reebe should now be admin spawnable
add: tropical lobtrosity
code: new config option for controlling overmap encounter default sizes
/:cl:

### CODE REVIEW

File: code/datums/components/largeobjecttransparency.dm [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	var/atom/par_atom = parent
+	par_atom.alpha = target_alpha
+	if(toggle_click)
+		par_atom.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/datum/component/largetransparency/proc/restoreAlpha()
+	var/atom/par_atom = parent
+	par_atom.alpha = initial_alpha
+	if(toggle_click)
+		par_atom.mouse_opacity = MOUSE_OPACITY_OPAQUE
```
Comment (ZephyrTFA) [inline code/datums/components/largeobjecttransparency.dm]: please ensure this is in snake_case not CamelCase or pascalCase

Comment (rye-rice) [reply code/datums/components/largeobjecttransparency.dm]: Woops, I should have checked first

**Comment by Latentish** [comment]
Very hyped for this.

**Comment by rye-rice** [comment]
it seems my rock walls are causing linters to break

**Comment by Latentish** [comment]
I've cleared up some of the TM's enough for this to TM. Once it's ready.

**Comment by rye-rice** [comment]
Conflicts fixed. Jungle ruins will be janky on launch however

**Comment by rye-rice** [comment]
testmerge i meant

**Comment by rye-rice** [comment]
finally passing linters at long long last

**Comment by rye-rice** [comment]
Conflicts fixed

**Comment by rye-rice** [comment]
conflcist sfixed

**Comment by rye-rice** [comment]
Conflicts fixed

**Comment by RDS88-beeman** [comment]
>Iron sand from rock planets doesn't contain the reagent Iron

why even mine.

**Comment by rye-rice** [comment]
e

**Comment by rye-rice** [comment]
conflists fixed

**Comment by rye-rice** [comment]
updated pr screenshots if you care

**Comment by rye-rice** [comment]
undrafted but just for you @triplezeta

**Comment by rye-rice** [comment]
i feel bad for whoever has to review this

File: code/__DEFINES/overmap.dm [hunk 2]
```diff
@@ -20,12 +20,16 @@
...
-#define DYNAMIC_WORLD_REEBE "reebe"
+
+#define DYNAMIC_WORLD_ROCKPLANET "rockplanet" //wacky planets
+#define DYNAMIC_WORLD_BEACHPLANET "beachplanet"
+#define DYNAMIC_WORLD_WASTEPLANET "wasteplanet"
+
+#define DYNAMIC_WORLD_REEBE "reebe" //celestial bodies
 #define DYNAMIC_WORLD_ASTEROID "asteroid"
 #define DYNAMIC_WORLD_SPACERUIN "space"
 #define DYNAMIC_WORLD_LIST_ALL list(DYNAMIC_WORLD_LAVA, DYNAMIC_WORLD_ICE, DYNAMIC_WORLD_SAND, DYNAMIC_WORLD_JUNGLE, DYNAMIC_WORLD_ROCKPLANET, DYNAMIC_WORLD_REEBE, DYNAMIC_WORLD_ASTEROID, DYNAMIC_WORLD_SPACERUIN)
```
Comment (ZephyrTFA) [inline code/__DEFINES/overmap.dm]: Add the new world types to this list

Comment (ZephyrTFA) [reply code/__DEFINES/overmap.dm]: And dont forget to add the options to the generate planet admin verb aswell

File: code/datums/mapgen/Cavegens/RockplanetCaves.dm [hunk 3]
```diff
@@ -1,139 +1,34 @@
...
+	closed_turf_types =  list(/turf/closed/mineral/random/asteroid/rockplanet = 1)
 
 	mob_spawn_chance = 2
 	flora_spawn_chance = 5
+	mob_spawn_chance = 3
+
+	mob_spawn_list = list(/mob/living/simple_animal/hostile/netherworld/asteroid = 30,
+		/mob/living/simple_animal/hostile/asteroid/fugu/asteroid = 30,
+		/mob/living/simple_animal/hostile/netherworld/migo/asteroid = 20, //mariuce
+//		/mob/living/simple_animal/hostile/ooze/grapes/asteroid = 20,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/Cavegens/RockplanetCaves.dm]: why is this commented out and not just removed

File: code/datums/mapgen/Cavegens/RockplanetCaves.dm [hunk 4]
```diff
@@ -1,139 +1,34 @@
+#define ROCKPLANET_DEFAULT_ATMOS "co2=95;n2=3;TEMP=210.15"
```
Comment (ZephyrTFA) [inline code/datums/mapgen/Cavegens/RockplanetCaves.dm]: this needs to be in a define file

**Comment by Latentish** [comment]
Aight bet. So the problem with testing these maps is that we have no way to spawn the new planets in game. Unless you add the ability to do so to the ruin/planet spawner, then review from maptainers is limited to viewing in an editor at best. Which, I'm willing to do for pretty much only this pr.

File: code/datums/mapgen/Cavegens/RockplanetCaves.dm [hunk 5]
```diff
@@ -1,139 +1,34 @@
...
-		/obj/structure/salvageable/server = 8,
+		/obj/structure/flora/rock = 3,
+		/obj/structure/flora/tree/cactus = 4,
+		/obj/structure/flora/ash/cacti = 1,
 	)
 
-	feature_spawn_list = list(/obj/structure/geyser/random = 1, /obj/effect/mine/shrapnel/human_only = 1)
-
-	initial_closed_chance = 45
+	feature_spawn_chance = 0
```
Comment (ZephyrTFA) [inline code/datums/mapgen/Cavegens/RockplanetCaves.dm:27]: if the feature spawn chance is zero why are you setting the list to anything other than null

File: code/datums/mapgen/Cavegens/WasteplanetCaves.dm [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		/obj/machinery/portable_atmospherics/canister/carbon_dioxide = 3,
+		/obj/structure/radioactive = 6,
+		/obj/structure/radioactive/stack = 6,
+		/obj/structure/radioactive/waste = 6,
+
+		/obj/structure/salvageable/machine = 20,
+		/obj/structure/salvageable/autolathe = 15,
+		/obj/structure/salvageable/computer = 10,
+		/obj/structure/salvageable/protolathe = 10,
+		/obj/structure/salvageable/circuit_imprinter = 8,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/Cavegens/WasteplanetCaves.dm:48]: No

File: code/datums/mapgen/biomes/beach_biomes.dm [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+		/obj/structure/flora/ausbushes/ywflowers = 1,
+		/obj/structure/flora/tree/palm = 1
+	)
+	flora_spawn_chance = 25
+	mob_spawn_list = list(
+		/mob/living/simple_animal/butterfly = 1,
+//		/mob/living/simple_animal/butterfly/beach = 1,
+		/mob/living/simple_animal/chicken/rabbit/normal = 1,
+//		/mob/living/simple_animal/chicken/beach = 1,
+//		/mob/living/simple_animal/chick/beach = 1,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/biomes/beach_biomes.dm]: Why are these commented out, remove them or add a reason for their commentation

File: code/datums/mapgen/biomes/rock_biomes.dm [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+		/obj/structure/flora/rock = 3,
+		/obj/structure/flora/tree/cactus = 4,
+		/obj/structure/flora/ash/cacti = 1,
+	)
+
+	mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/netherworld/asteroid = 30,
+		/mob/living/simple_animal/hostile/asteroid/fugu/asteroid = 30,
+		/mob/living/simple_animal/hostile/netherworld/migo/asteroid = 20, //mariuce
+//		/mob/living/simple_animal/hostile/ooze/grapes/asteroid = 20,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/biomes/rock_biomes.dm]: and here

File: code/datums/mapgen/biomes/rock_biomes.dm [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+/datum/biome/rock/wetlands
+	open_turf_types = list(/turf/open/floor/plating/asteroid/rockplanet/wet/lit = 1)
+	flora_spawn_chance = 5
+	mob_spawn_chance = 4
+	mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/netherworld/asteroid = 30,
+		/mob/living/simple_animal/hostile/asteroid/fugu/asteroid = 30,
+		/mob/living/simple_animal/hostile/netherworld/migo/asteroid = 20, //mariuce
+//		/mob/living/simple_animal/hostile/ooze/grapes/asteroid = 20,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/biomes/rock_biomes.dm]: and here

File: code/datums/mapgen/biomes/rock_biomes.dm [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+		/obj/structure/flora/ash/fern = 2,
+		/obj/structure/flora/ash/puce = 1,
+	)
+	feature_spawn_list = list(/obj/structure/geyser/random = 1, /obj/structure/elite_tumor = 2)
+	mob_spawn_chance = 4
+	mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/netherworld/asteroid = 20,
+		/mob/living/simple_animal/hostile/asteroid/fugu/asteroid = 20,
+		/mob/living/simple_animal/hostile/netherworld/migo/asteroid = 20, //mariuce
+//		/mob/living/simple_animal/hostile/ooze/grapes/asteroid = 20,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/biomes/rock_biomes.dm]: and here

File: code/datums/mapgen/biomes/rock_biomes.dm [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+		/obj/structure/flora/rock = 4,
+		/obj/structure/flora/rock/pile = 4,
+		/obj/structure/flora/ash/fern = 4,
+		/obj/structure/flora/ash/puce = 2,
+	)
+	mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/netherworld/asteroid = 30,
+		/mob/living/simple_animal/hostile/asteroid/fugu/asteroid = 30,
+		/mob/living/simple_animal/hostile/netherworld/migo/asteroid = 20, //mariuce
+//		/mob/living/simple_animal/hostile/ooze/grapes/asteroid = 20,
```
Comment (ZephyrTFA) [inline code/datums/mapgen/biomes/rock_biomes.dm]: also here

File: code/datums/ruins/rockplanet.dm [hunk 12]
```diff
@@ -24,24 +17,19 @@
...
-	suffix = "rockplanet_clowncrash.dmm"
-
 /datum/map_template/ruin/rockplanet/cult
 	name = "Cult base"
 	id = "rockcult"
 	description = "Cultists seem to have gotten here too."
 	suffix = "rockplanet_cult.dmm"
 
-
+/* TODO: GREEBLE
```
Comment (ZephyrTFA) [inline code/datums/ruins/rockplanet.dm:26]: todo

File: code/game/objects/effects/decals/turfdecal/flooring_decals.dm [hunk 13]
```diff
@@ -110,6 +110,12 @@
 	}													\
 	/obj/effect/turf_decal/spline/fancy/corner/##color_name {	\
 		icon_state = "spline_fancy_corner"				\
+	}													\
+	/obj/effect/turf_decal/road/line/##color_name {		\
+		color = ##tile_color							\
+	}													\
+	/obj/effect/turf_decal/road/line/edge/##color_name {		\
+		color = ##tile_color							\
```
Comment (ZephyrTFA) [inline code/game/objects/effects/decals/turfdecal/flooring_decals.dm:118]: holy damn that formatting is beyond awful. The formatting was already there so its whatever; but whoever started it should be taken outback

File: code/game/turfs/open/acid.dm [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+	footstep = FOOTSTEP_LAVA
+	barefootstep = FOOTSTEP_LAVA
+	clawfootstep = FOOTSTEP_LAVA
+	heavyfootstep = FOOTSTEP_LAVA
+	density = TRUE //This will prevent hostile mobs from pathing into acid, while the canpass override will still let it function like an open turf
+
+/turf/open/acid/CanAllowThrough(atom/movable/AM, turf/target)
+	. = ..()
+	return TRUE
+
```
Comment (ZephyrTFA) [inline code/game/turfs/open/acid.dm]:
```suggestion

/turf/open/acid/CanAllowThrough(atom/movable/AM, turf/target)
	if(ishostile(AM))
		return FALSE
	return ..()

```
No don't do it like that, do it like this

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]
Looks good for the most part, haven't reviewed map changes because I don't hate myself enough to do that

**Review by Latentish — CHANGES_REQUESTED** [review]
wastplanet_clowncrash - Has 10 individual plasma sheets instead of a 10 stack.
wasteplanet_pod -  Might be better as a greeble, but doesn't need to be blacklisted right now (no change needed on this one)
wasteplanet_solgov_terraformingfacility - Improper use of area passthrough, large space taken up by weird turf and area passthrough, use of /turf, inappropriate mapsize for the ruin itself, not much to the ruin itself, rogue space tile. Just remove this one.
wasteplanet_tarpit - Use of airlock cycler helper on normal door. Bit of a lootbox with no threat,, 2 ATVs, a cr20, 2 pistols, a mosin, an esword, and a hand drill. Cut back on look very heavily.
wasteplanet_tradepost - Bit of a lootbox, fully functional cargo ripley, 2 colt detective specials, antique laser. Uses lots of singular plasma sheets. Remove 1 colt and the antique laser, and replace the plasma with proper stacks.

beach_ancient_ruin - use of bwoink mines. I'd rather these be abus only honestly.
beach_sandstone_castle - items on top of turf/area passthroughs (this sometimes causes space tiles to spawn), not much there besides trash spawns and a trophy case. Fix those passthrough objects at minimum or remove it.

rockplanet_clock - Contains a bronze golem carbon mob, this will do nothing but stand there as carbons have no AI. Add adamantine or a golem frame instead.
rockplanet_crash_cult - use of airlock cyclelink helpers (these don't work to my knowledge, and a functioning airlock is superfluous in this case.
rockplanet_dangerpod - This ruin isn't much, it'd be ok as a greeble, but otherwise is basically just a pod with an enemy in it with almost no loot. This map should be blacklisted on the tgs.
rockplanet_daniel - I find this ruin unsavory as it's just a suicide cabin. It's not much of a storytelling other than "person killed themselves on their birthday". Suicide in environmental storytelling can be done well in instances of larger maps with lots of other storytelling, but this is not done well. Remove the suicide mention and put something else here, or remove this one please.
rockplanet_house - Why there is more than one ruin that is literally nothing but "person in a house who committed suicide" I don't know. But just remove this one.
rockplanet_pioneer - This is a funny reference but it's barely anything. Remove or blacklist this for greebles.
rockplanet_rd_god - This is a category of ruin that we as maptainers have decided we don't really want. Which is a ruin that is nothing but tiles and an enemy. It might have some loot but this isn't enough. Remove this one please.

File: code/game/objects/effects/decals/turfdecal/flooring_decals.dm (hunk 13 above)
Comment (MarkSuckerberg) [reply code/game/objects/effects/decals/turfdecal/flooring_decals.dm:118]: hi, here and waiting to be taken out behind the shed

although seriously that's just how defines gotta be, or rather, it's the most readable way of having them

Comment (ZephyrTFA) [reply code/game/objects/effects/decals/turfdecal/flooring_decals.dm:118]: 🔫

Comment (rye-rice) [reply code/game/objects/effects/decals/turfdecal/flooring_decals.dm:118]: Rest in peace mark

File: code/datums/mapgen/Cavegens/WasteplanetCaves.dm (hunk 6 above)
Comment (rye-rice) [reply code/datums/mapgen/Cavegens/WasteplanetCaves.dm:48]: About this, you cant actually get imprinters or protolathes from this, they are just the names of the structures

**Comment by rye-rice** [comment]
>cat noises

A lot of the ruins were commented out here with the comments "GREEBLE" or "minor ruin"
Also, ruins here were unchanged from  how they were before, but they are marked as added because of renaming to fit the new planets or changing of the planetary turfs

**Comment by rye-rice** [comment]
applied mapping suggestions, removed 2 ruins

**Review by Latentish — CHANGES_REQUESTED** [review]
- Remove rockplanet_rd_god.dmm
- Fix areas in wasteplanet_lab.dmm
- Fix areas in wasteplanet_ore_processing_facility.dmm

**Review by Latentish — APPROVED** [review]
I am Jane Shiptest and I approve this testering for a fucking merge.

File: code/datums/components/largeobjecttransparency.dm [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+	var/atom/par_atom = parent
+	par_atom.alpha = target_alpha
+	if(toggle_click)
+		par_atom.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/datum/component/largetransparency/proc/restore_alpha()
+	var/atom/par_atom = parent
+	par_atom.alpha = initial_alpha
+	if(toggle_click)
+		par_atom.mouse_opacity = MOUSE_OPACITY_ICON
```
Comment (Fikou) [inline code/datums/components/largeobjecttransparency.dm:111]: initial here please

File: code/datums/mapgen/Cavegens/WasteplanetCaves.dm [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+		/mob/living/simple_animal/hostile/hivebot/rapid/rockplanet = 30,
+		//crystal mobs, very difficult
+		/mob/living/simple_animal/hostile/asteroid/goliath/beast/ancient/crystal = 1,
+		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/forgotten = 1,
+		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/crystal = 1,
+		//bots, are hostile
+		/mob/living/simple_animal/bot/firebot/rockplanet = 15,
+		/mob/living/simple_animal/bot/secbot/ed209/rockplanet = 3,
+		/mob/living/simple_animal/hostile/abandoned_minebot = 15,
+		/mob/living/simple_animal/bot/floorbot/rockplanet = 15)
```
Comment (Fikou) [inline code/datums/mapgen/Cavegens/WasteplanetCaves.dm]: leave a trailing comma and leave the parentheses unindented once on next line on all of these please

File: code/datums/mapgen/biomes/beach_biomes.dm [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+		/obj/structure/flora/ausbushes/palebush = 1,
+		/obj/structure/flora/ausbushes/pointybush = 1,
+		/obj/structure/flora/ausbushes/ppflowers = 1,
+		/obj/structure/flora/ausbushes/reedbush = 1,
+		/obj/structure/flora/ausbushes/sparsegrass = 1,
+		/obj/structure/flora/ausbushes/stalkybush = 1,
+		/obj/structure/flora/ausbushes/stalkybush = 1,
+		/obj/structure/flora/ausbushes/sunnybush = 1,
+		/obj/structure/flora/ausbushes/ywflowers = 1,
+		/obj/structure/flora/tree/palm = 1
```
Comment (Fikou) [inline code/datums/mapgen/biomes/beach_biomes.dm]: trailing comma on these

File: code/datums/mapgen/biomes/lava_biomes.dm [hunk 18]
```diff
@@ -0,0 +1,159 @@
+/datum/biome/lavaland
+	open_turf_types = list(/turf/open/floor/plating/asteroid/basalt/lava_land_surface/lit = 1)
+	flora_spawn_chance = 1
+	flora_spawn_list = list(
+		/obj/structure/flora/ausbushes/ywflowers/hell = 10,
+		/obj/structure/flora/ausbushes/sparsegrass/hell = 40,
+		/obj/structure/flora/ash/fern = 5,
+		/obj/structure/flora/ash/fireblossom = 1,
+		/obj/structure/flora/ash/puce = 5
```
Comment (Fikou) [inline code/datums/mapgen/biomes/lava_biomes.dm]: and here

File: code/datums/mapgen/biomes/snow_biomes.dm [hunk 19]
```diff
@@ -0,0 +1,190 @@
+
+/datum/biome/snow
+	open_turf_types = list(/turf/open/floor/plating/asteroid/snow/lit = 25)
+	flora_spawn_list = list(
+		/obj/structure/flora/tree/pine = 2,
+		/obj/structure/flora/rock/icy = 2,
+		/obj/structure/flora/rock/pile/icy = 2,
+		/obj/structure/flora/grass/both = 6,
+		/obj/structure/flora/ash/chilly = 2
```
Comment (Fikou) [inline code/datums/mapgen/biomes/snow_biomes.dm]: and here

File: code/datums/mapgen/planets/_planet.dm [hunk 20]
```diff
@@ -0,0 +1,73 @@
+/datum/planet
+	var/list/biomes = list(
+		"coldest" = list(
+			"biome_lowest_humidity" = /datum/biome/grass,
```
Comment (Fikou) [inline code/datums/mapgen/planets/_planet.dm]: are these not defines? they need to be defines

File: code/datums/mapgen/planets/_planet.dm [hunk 21]
```diff
@@ -0,0 +1,73 @@
+/datum/planet
+	var/list/biomes = list(
+		"coldest" = list(
```
Comment (Fikou) [inline code/datums/mapgen/planets/_planet.dm]: make these defines too if you will

File: code/game/turfs/open/acid.dm [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+	light_power = 0.75
+	light_color = LIGHT_COLOR_SLIME_LAMP
+	bullet_bounce_sound = 'sound/items/welder2.ogg'
+
+	footstep = FOOTSTEP_LAVA
+	barefootstep = FOOTSTEP_LAVA
+	clawfootstep = FOOTSTEP_LAVA
+	heavyfootstep = FOOTSTEP_LAVA
+
+/turf/open/acid/CanAllowThrough(atom/movable/AM, turf/target)
```
Comment (Fikou) [inline code/game/turfs/open/acid.dm:18]: use a more readable var name than AM

File: code/game/turfs/open/acid.dm [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+	underlay_appearance.icon = 'icons/turf/floors.dmi'
+	underlay_appearance.icon_state = "basalt"
+	return TRUE
+
+/turf/open/acid/attackby(obj/item/C, mob/user, params)
+	..()
+	if(istype(C, /obj/item/stack/rods))
+		var/obj/item/stack/rods/R = C
+		var/obj/structure/lattice/H = locate(/obj/structure/lattice, src)
+		if(H)
```
Comment (Fikou) [inline code/game/turfs/open/acid.dm:85]: all of these vars need to be readable

File: code/game/turfs/open/acid.dm [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+/turf/open/acid/proc/melt_stuff(AM)
+	if(is_safe_to_cross())
+		return FALSE
+	. = FALSE
+	var/thing_to_check = src
+	if (AM)
+		thing_to_check = list(AM)
+	for(var/thing in thing_to_check)
+		if(isobj(thing))
+			var/obj/O = thing
```
Comment (Fikou) [inline code/game/turfs/open/acid.dm:115]: same here

File: code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm [hunk 25]
```diff
@@ -73,10 +73,7 @@
 		var/mob/living/L = target
 		if(prob(80))
 			var/atom/throw_target = get_edge_target_turf(L, dir)
-			L.throw_at(throw_target, rand(1,2), 7, src)
-		else
-			L.Paralyze(20)
-			visible_message("<span class='danger'>[src] knocks [L] down!</span>")
+			L.throw_at(throw_target, rand(1,2), 7, src, gentle = TRUE)
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm:76]: do you need that prob then

File: code/modules/mob/living/simple_animal/hostile/abandoned_minebot.dm [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+/obj/effect/spawner/lootdrop/minebot
+	loot = list(/obj/item/borg/upgrade/modkit/minebot_passthrough = 15,
+				/obj/item/borg/upgrade/modkit/chassis_mod = 15,
+				/obj/item/borg/upgrade/modkit/tracer = 15,
+				/obj/item/borg/upgrade/modkit/cooldown = 6,
+				/obj/item/borg/upgrade/modkit/damage = 6,
+				/obj/item/borg/upgrade/modkit/range = 6,
+				/obj/item/borg/upgrade/modkit/aoe/mobs = 6,
+				/obj/item/borg/upgrade/modkit/aoe/turfs = 6,
+				/obj/item/borg/upgrade/modkit/trigger_guard = 6)
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/abandoned_minebot.dm:54]: reformat this list please

File: code/modules/mob/living/simple_animal/friendly/beachcarp.dm [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "punches"
+	response_harm_simple = "punch"
+	stop_automated_movement = 1
+	friendly_verb_continuous = "nibbles"
+	friendly_verb_simple = "nibble"
+	turns_per_move = 2
+
+
+/mob/living/simple_animal/beachcarp/Life()
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/friendly/beachcarp.dm:28]: is the only thing this is for, moving onto only water turfs? if yes, i am really not a fan of this solution

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by ZephyrTFA** [comment]
I want this merged, I'm tired of it always being test merged. I will accept the current existing technical debt

**Comment by rye-rice** [comment]
um.. i wasnt expecting that, ill go address the reviews in another pr then

**Comment by Zevotech** [comment]
worth it, let the maps flow freely

**Pull request merged**